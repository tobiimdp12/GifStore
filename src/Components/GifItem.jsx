import React from "react";
import ButtonCard from "./Buttons/ButtonCard/ButtonCard";
import Loader from "./Loader/Loader";
import ButtonDownload from "./Buttons/ButtonDownload/ButtonDownload";
import LikedGif from "../Helpers/LikedGif";
import Swal from "sweetalert2";
import 'animate.css';


function GifItem({ gif }) {

  const handleModal = () => {
    Swal.fire({
      imageUrl: gif.url,
      imageWidth: 400,
      imageHeight: "80%",
      imageAlt: "Custom image",
      color: "#fff",
      background: "#000",
      backdrop: "#0000008a",
    });
  };

  let isLiked = LikedGif(gif);//-

  return (
    <div>
      {!gif.url ? (
        <Loader />
      ) : (
        <div>
          <div className="card animate__animated animate__fadeInTopRight   animate__slow">
            <ButtonDownload gif={gif} />
            <ButtonCard gif={gif} handleLiked={isLiked} />
            <img src={gif.url} alt="gif" onClick={handleModal} role="img" />
            <p>{gif.title}</p>
          </div>
        </div>
      )}
    </div>
  );
}

export default GifItem;
