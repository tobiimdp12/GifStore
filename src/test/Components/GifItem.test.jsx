import { render } from "@testing-library/react";
import GifItem from "../../Components/GifItem";

describe("Test en <GifItem/>", () => {
  
  const title = "Dragon Ball";
  const url = "https://dragon-ball.com/goku.jpg";
  const gif = { title: title, url: url };

  test("Debe coincidir con el snapshot", () => {
    console.log(gif);
    const { container } = render(<GifItem gif={gif} />);
    expect(container).toMatchSnapshot();

  });

  test('Debe mostrar URL y ALT', () => {
        
    render( <GifItem title={title} url={url} /> )
â€‹
    // screen.debug();
       
    const {src, alt} = screen.getByRole( 'img' );
    expect( src ).toContain(url);
    expect( alt ).toContain('gif');
});
});
