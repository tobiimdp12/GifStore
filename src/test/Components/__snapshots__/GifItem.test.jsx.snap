// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <GifItem/> Debe coincidir con el snapshot 1`] = `
<div>
  <div>
    <div>
      <div
        class="card animate__animated animate__fadeInTopRight   animate__slow"
      >
        <a
          class="download"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M624 706.3h-74.1V464c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v242.3H400c-6.7 0-10.4 7.7-6.3 12.9l112 141.7a8 8 0 0 0 12.6 0l112-141.7c4.1-5.2.4-12.9-6.3-12.9z"
            />
            <path
              d="M811.4 366.7C765.6 245.9 648.9 160 512.2 160S258.8 245.8 213 366.6C127.3 389.1 64 467.2 64 560c0 110.5 89.5 200 199.9 200H304c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8h-40.1c-33.7 0-65.4-13.4-89-37.7-23.5-24.2-36-56.8-34.9-90.6.9-26.4 9.9-51.2 26.2-72.1 16.7-21.3 40.1-36.8 66.1-43.7l37.9-9.9 13.9-36.6c8.6-22.8 20.6-44.1 35.7-63.4a245.6 245.6 0 0 1 52.4-49.9c41.1-28.9 89.5-44.2 140-44.2s98.9 15.3 140 44.2c19.9 14 37.5 30.8 52.4 49.9 15.1 19.3 27.1 40.7 35.7 63.4l13.8 36.5 37.8 10C846.1 454.5 884 503.8 884 560c0 33.1-12.9 64.3-36.3 87.7a123.07 123.07 0 0 1-87.6 36.3H720c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h40.1C870.5 760 960 670.5 960 560c0-92.7-63.1-170.7-148.6-193.3z"
            />
          </svg>
        </a>
        <div>
          <a
            role="button"
          >
            <svg
              class="heart active"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
              />
              <path
                d="M679.7 201c-73.1 0-136.5 40.8-167.7 100.4C480.8 241.8 417.4 201 344.3 201c-104 0-188.3 82.6-188.3 184.5 0 201.2 356 429.3 356 429.3s356-228.1 356-429.3C868 283.6 783.7 201 679.7 201z"
              />
            </svg>
          </a>
        </div>
        <img
          alt="gif"
          src="https://dragon-ball.com/goku.jpg"
        />
        <p>
          Dragon Ball
        </p>
      </div>
    </div>
  </div>
</div>
`;
